File: app/src/main/java/com/esbati/keivan/persiancalendar/Features/Notification/NotificationBroadcastReceiver.java
package com.esbati.keivan.persiancalendar.Features.Notification;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.esbati.keivan.persiancalendar.Utils.AndroidUtilities;

/**
 * Created by Keivan Esbati on 4/10/2017.
 */

public class NotificationBroadcastReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String event = intent.getAction();
        Log.d(getClass().getSimpleName(), "Received intent with action " + event);

        //Update Notification
        NotificationUpdateService.enqueueUpdate(context);

        //Start Application Service if Not Running
        if (!AndroidUtilities.isServiceRunning(ApplicationService.class))
            context.startService(new Intent(context, ApplicationService.class));
    }
}

File: app/src/main/java/com/esbati/keivan/persiancalendar/features/notification/NotificationBroadcastReceiver.kt
package com.esbati.keivan.persiancalendar.features.notification

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

/**
 * Created by Keivan Esbati on 4/10/2017.
 */

class NotificationBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        Log.d(javaClass.simpleName, "Received intent with action $action")

        // In case of following event proceed to update notification:
        // Intent.ACTION_BOOT_COMPLETED
        // Intent.ACTION_DATE_CHANGED
        // Intent.ACTION_TIME_CHANGED
        // Intent.ACTION_TIMEZONE_CHANGED
        // Intent.ACTION_TIME_TICK
        NotificationUpdateService.enqueueUpdate(context)
    }
}
